Rudder/RSEventRepository.m

```diff
@@ -140,33 +148,47 @@ - (void) __initiateFactories : (NSArray*) destinations {
             for (RSServerDestination *destination in destinations) {
                 [destinationDict setObject:destination forKey:destination.destinationDefinition.displayName];
             }
-            NSMutableDictionary<NSString*, id<RSIntegration>> *tempIntegrationOpDict = [[NSMutableDictionary alloc] init];
             for (id<RSIntegrationFactory> factory in self->config.factories) {
                 RSServerDestination *destination = [destinationDict objectForKey:factory.key];
                 if (destination != nil && destination.isDestinationEnabled == YES) {
                     NSDictionary *destinationConfig = destination.destinationConfig;
                     if (destinationConfig != nil) {
                         id<RSIntegration> nativeOp = [factory initiate:destinationConfig client:[RSClient sharedInstance] rudderConfig:self->config];
                         [RSLogger logDebug:[[NSString alloc] initWithFormat:@"Initiating native SDK factory %@", factory.key]];
-                        [tempIntegrationOpDict setValue:nativeOp forKey:factory.key];
+                        [integrationOperationMap setValue:nativeOp forKey:factory.key];
                         [RSLogger logDebug:[[NSString alloc] initWithFormat:@"Initiated native SDK factory %@", factory.key]];
                         // put native sdk initialization callback
                     }
                 }
             }
-            self->integrationOperationMap = tempIntegrationOpDict;
         }
-        self->isFactoryInitialized = YES;
-        @synchronized (self->eventReplayMessage) {
-            [RSLogger logDebug:@"replaying old messages with factory"];
-            NSArray *tempMessages = [self->eventReplayMessage copy];
-            if (tempMessages.count > 0) {
-                for (RSMessage *msg in tempMessages) {
-                    [self makeFactoryDump:msg];
-                }
+    }
+}
+
+- (void) __initiateCustomFactories {
+    if (self->config == nil || config.customFactories == nil || config.customFactories.count == 0) {
+        [RSLogger logInfo:@"EventRepository: initiateCustomFactories: No custom factory found"];
+        return;
+    }
+    for (id<RSIntegrationFactory> factory in self->config.customFactories) {
+        id<RSIntegration> nativeOp = [factory initiate:@{} client:[RSClient sharedInstance] rudderConfig:self->config];
+        [RSLogger logDebug:[[NSString alloc] initWithFormat:@"Initiating custom factory %@", factory.key]];
+        [self->integrationOperationMap setValue:nativeOp forKey:factory.key];
+        [RSLogger logDebug:[[NSString alloc] initWithFormat:@"Initiated custom SDK factory %@", factory.key]];
+        // put custom sdk initalization callback
+    }
+}
+
+- (void) __replayMessageQueue {
+    @synchronized (self->eventReplayMessage) {
+        [RSLogger logDebug:@"replaying old messages with factory"];
+        NSArray *tempMessages = [self->eventReplayMessage copy];
```

Comment:
Why you need this tempMessages variable here? I can see this is read only variable, not needed.

Reply:
I've made the necessary changes, please review it again. If any further optimisation need to be done, I'll implement it.